File: 4lab/1/stack.py
fin = open("stack.in", "r")
fout = open("stack.out", "w")

stack = []
n = int(fin.readline())

for line in fin:
	if (line[0] == '+'):
		stack.append(int(line.split("+ ")[-1]))
	else:
		fout.write(str(stack[-1])+'\n')
		stack.pop()
fin.close()
fout.close()

File: 4lab/1/stack.cpp
#include <fstream>
#include <iostream>
#include <cstring>
 
using namespace std;
 
 
int main() {
    int n, m, q, cnt = 0;
    char c;
    ifstream fin("stack.in");
    ofstream fout("stack.out");
     
    fin >> n;
 
    int a[n];
     
    for (int i = 0; i < n; i++)
    {
        fin >> c;
         
        if (c == '+')
            fin >> a[cnt++];
        else
            fout << a[--cnt] << "\n";
             
    }
 
    fin.close();
    fout.close();
    return 0;
}
